Library/Homebrew/cask/tab.rb

```diff
@@ -0,0 +1,184 @@
+# typed: true
+# frozen_string_literal: true
+
+module Cask
+  class Tab
+    extend Cachable
+
+    FILENAME = "INSTALL_RECEIPT.json"
+
+    attr_accessor :homebrew_version, :tabfile, :loaded_from_api, :installed_as_dependency, :installed_on_request,
+                  :time, :dependencies, :arch, :source, :installed_on, :artifacts
+
+    # Instantiates a {Tab} for a new installation of a cask.
+    sig { params(cask: Cask).returns(Tab) }
+    def self.create(cask)
+      attributes = {
+        "homebrew_version"        => HOMEBREW_VERSION,
+        "tabfile"                 => cask.metadata_main_container_path/FILENAME,
+        "loaded_from_api"         => cask.loaded_from_api?,
+        "installed_as_dependency" => false,
+        "installed_on_request"    => false,
+        "time"                    => Time.now.to_i,
+        "dependencies"            => Tab.runtime_deps_hash(cask, cask.depends_on),
+        "arch"                    => Hardware::CPU.arch,
+        "source"                  => {
+          "path"         => cask.sourcefile_path.to_s,
+          "tap"          => cask.tap&.name,
+          "tap_git_head" => nil, # Filled in later if possible
+          "version"      => cask.version.to_s,
+        },
+        "installed_on"            => DevelopmentTools.build_system_info,
+        "artifacts"               => cask.to_h["artifacts"],
```

Comment:
Question: Why are we including the artifacts here?

Reply:
In https://github.com/Homebrew/brew/issues/17013#issuecomment-2158659721 it was suggested to have the uninstall information available here, and adding all of the artifacts here was an easy way to do that. We could also filter the artifacts to only include uninstall-related things if preferred.

I think the basic thinking is that uninstalling the cask should use this info rather than the newest cask information which might not apply

Reply:
That sounds reasonable. I think filtering it down might be good if it's simple though.